src/wp-includes/blocks.php

```diff
@@ -1325,7 +1325,8 @@ function build_comment_query_vars_from_block( $block ) {
 			} elseif ( 'oldest' === $default_page ) {
 				$comment_args['paged'] = 1;
 			} elseif ( 'newest' === $default_page ) {
-				$comment_args['paged'] = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
+				$max_num_pages         = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
+				$comment_args['paged'] = 0 === $max_num_pages ? 1 : $max_num_pages;
```

Comment:
What is this patch doing? When there are no comments, it sets the `'paged'` arg in the comments query to the default value of `1`. 

Should it be setting the default value? Or should it _not set_ `'paged'` arg and then let Core natively handle setting the default value. Setting here means that if the default value changes someday, it'll need to be changed in multiple places. cc @dream-encode what do you think?

Another way to handle this could be:

```suggestion
				if ( 0 === $max_num_pages ) {
					$comment_args['paged'] = $max_num_pages;
				}
```

Reply:
Note on landing this PR in time for RC1:
This fix is needed for 6.0 as the issue was introduced in 6.0's cycle and its severity is severe. The patch works and can in as-is to fix the issue. The question about whether to set a default value or not set `'paged'` arg can be handled separately if more time is needed for discussion and consensus.